test/capi/testPaint.cpp

```diff
@@ -0,0 +1,113 @@
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <thorvg_capi.h>
+#include "catch.hpp"
+
+TEST_CASE("Paint Transformation", "[capiPaintTransformation]")
+{
+    Tvg_Paint *paint = NULL;
+    Tvg_Matrix matrix_set = {1, 0, 0, 0, 1, 0, 0, 0, 1}, matrix_get;
+
+    paint = tvg_shape_new();
+    REQUIRE(paint != NULL);
+
+    REQUIRE(tvg_paint_transform(paint, &matrix_set) == TVG_RESULT_SUCCESS);
+
+    REQUIRE(tvg_paint_scale(paint, 2.0f) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_scale(paint, 0.5f) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_rotate(paint, 180.0f) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_rotate(paint, 180.0f) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_translate(paint, 10.0f, 10.0f) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_translate(paint, -10.0f, -10.0f) == TVG_RESULT_SUCCESS);
+
+    REQUIRE(tvg_paint_get_transform(paint, &matrix_get) == TVG_RESULT_SUCCESS);
+
+    REQUIRE(matrix_get.e11 == matrix_set.e11);
+    REQUIRE(matrix_get.e12 == matrix_set.e12);
+    REQUIRE(matrix_get.e13 == matrix_set.e13);
+    REQUIRE(matrix_get.e21 == matrix_set.e21);
+    REQUIRE(matrix_get.e22 == matrix_set.e22);
+    REQUIRE(matrix_get.e23 == matrix_set.e23);
+    REQUIRE(matrix_get.e31 == matrix_set.e31);
+    REQUIRE(matrix_get.e32 == matrix_set.e32);
+    REQUIRE(matrix_get.e33 == matrix_set.e33);
+
+    REQUIRE(tvg_paint_del(paint) == TVG_RESULT_SUCCESS);
+}
+
+TEST_CASE("Paint Opacity", "[capiPaintOpacity]")
+{
+    Tvg_Paint *paint = NULL;
+    uint8_t opacity_set, opacity_get;
+
+    paint = tvg_shape_new();
+    REQUIRE(paint != NULL);
+
+    opacity_set = 0;
+    REQUIRE(tvg_paint_set_opacity(paint, opacity_set) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_get_opacity(paint, &opacity_get) == TVG_RESULT_SUCCESS);
+    REQUIRE(opacity_get == opacity_get);
+
+    opacity_set = 128;
+    REQUIRE(tvg_paint_set_opacity(paint, opacity_set) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_get_opacity(paint, &opacity_get) == TVG_RESULT_SUCCESS);
+    REQUIRE(opacity_get == opacity_get);
+
+    opacity_set = 255;
+    REQUIRE(tvg_paint_set_opacity(paint, opacity_set) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_get_opacity(paint, &opacity_get) == TVG_RESULT_SUCCESS);
+    REQUIRE(opacity_get == opacity_get);
+
+    REQUIRE(tvg_paint_del(paint) == TVG_RESULT_SUCCESS);
+}
+
+TEST_CASE("Paint Bounds", "[capiPaintBounds]")
+{
+    Tvg_Paint *paint = NULL;
+    float x = 0, y = 10, w = 100, h = 100;
```

Comment:
I wouldn't use the same values as the args, so in case something is mixed in the code, we can catch it

Reply:
Changed `w` value